weasyl/controllers/general.py

```diff
@@ -103,9 +108,9 @@ def search_(request):
             None,
             # Search results
             {
-                "submit": search.browse(request.userid, rating, 22, form, find="submit"),
-                "char": search.browse(request.userid, rating, 22, form, find="char"),
-                "journal": search.browse(request.userid, rating, 22, form, find="journal"),
+                "submit": search.browse(request.userid, rating, 22, "submit", cat, backid, nextid),
```

Comment:
I think this could benefit from named arguments, and `backid`/`nextid` being parsed in advance.

Reply:
Is this what you were thinking? https://github.com/sl1-1/weasyl/tree/remove-storify-from-controllers-general
Not sure if there is a way to change a pullrequest, or if you should pull the changes to your branch? :)